cfme/tests/openstack/cloud/test_networks.py

```diff
@@ -0,0 +1,216 @@
+    collection = appliance.collections.network_routers
+    router = collection.create(name=fauxfactory.gen_alpha(),
+                               tenant=provider.get_yaml_data()['tenant'],
+                               provider=provider,
+                               network_manager='{} Network Manager'.format(provider.name),
+                               has_external_gw=ext_gw,
+                               ext_network=ext_network,
+                               ext_network_subnet=ext_subnet)
+    return router
+
+
+@pytest.yield_fixture(scope='function')
+def network(provider, appliance):
+    """Create cloud network"""
+    network = create_network(appliance, provider, is_external=False)
+    yield network
+    delete_entity(network)
+
+
+@pytest.yield_fixture(scope='function')
+def ext_network(provider, appliance):
+    """Create external cloud network"""
+    network = create_network(appliance, provider, is_external=True)
+    yield network
+    delete_entity(network)
+
+
+@pytest.fixture(scope='module')
+def subnet_cidr():
+    return '11.11.11.0/24'
+
+
+@pytest.yield_fixture(scope='function')
+def subnet(provider, appliance, network, subnet_cidr):
+    """Creates subnet for the given network"""
+    subnet = create_subnet(appliance, provider, network, subnet_cidr)
+    yield subnet
+    delete_entity(subnet)
+
+
+@pytest.yield_fixture(scope='function')
+def ext_subnet(provider, appliance, ext_network, subnet_cidr):
+    """Creates subnet for the given external network"""
+    subnet = create_subnet(appliance, provider, ext_network, subnet_cidr)
+    yield subnet
+    delete_entity(subnet)
+
+
+@pytest.yield_fixture(scope='function')
+def router(provider, appliance):
+    """Creates network router"""
+    router = create_router(appliance, provider, ext_gw=False)
+    yield router
+    delete_entity(router)
+
+
+@pytest.yield_fixture(scope='function')
+def router_with_gw(provider, appliance, ext_subnet):
+    """Creates network router with external network as a gateway"""
+    router = create_router(appliance, provider, ext_gw=True, ext_network=ext_subnet.network,
+                           ext_subnet=ext_subnet.name)
+    yield router
+    delete_entity(router)
+
+
+def test_create_network(network, provider):
+    assert network.exists
+    assert network.parent_provider.name == provider.name
+    assert network.cloud_tenant == provider.get_yaml_data()['tenant']
+
+
+def test_edit_network(network):
+    network.edit(name=fauxfactory.gen_alpha())
+    wait_for(network.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
+             delay=10)
+    network.browser.refresh()
+    assert network.exists
+
+
+def test_delete_network(network, appliance):
+    network.delete()
+    wait_for(network.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
+             delay=10)
+    navigate_to(appliance.collections.cloud_networks, 'All')
```

Comment:
**Question:** Why do we need this navigation?

Reply:
Deleting..